smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/model/TestEquatableConformanceTransformer.kt

```diff
@@ -0,0 +1,63 @@
+package software.amazon.smithy.swift.codegen.model
+
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.shapes.ShapeId
+import software.amazon.smithy.model.shapes.ShapeType
+import software.amazon.smithy.model.shapes.StructureShape
+import software.amazon.smithy.model.transform.ModelTransformer
+import software.amazon.smithy.protocoltests.traits.HttpResponseTestsTrait
+import software.amazon.smithy.swift.codegen.customtraits.TestEquatableConformanceTrait
+
+object TestEquatableConformanceTransformer {
```

Comment:
A new model transformer that applies the `TestEquatableConformanceTrait` to any shape that is used in a response protocol test, or any shape nested within such a shape.

See inline comments for greater detail.